File: backend/src/main/java/vn/gtel/pm2/sizing/dto/request/UpdateProjectAssumptionRequest.java
package vn.gtel.pm2.sizing.dto.request;

import jakarta.validation.constraints.NotNull;
import jakarta.validation.constraints.PositiveOrZero;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
public class UpdateProjectAssumptionRequest {

    @PositiveOrZero
    @NotNull
    private Integer concurrentUsers;

    @PositiveOrZero
    @NotNull
    private Float headroom;

    @PositiveOrZero
    @NotNull
    private Float requestsPerUserPerSecond;

    @PositiveOrZero
    @NotNull
    private Float apiCallsPerRequest;

    @PositiveOrZero
    @NotNull
    private Float dbRatioPerRequest;

    @PositiveOrZero
    @NotNull
    private Float searchRatioPerRequest;

    @PositiveOrZero
    @NotNull
    private Float cacheRatioPerRequest;

    @PositiveOrZero
    @NotNull
    private Float kafkaRatioPerRequest;

    @PositiveOrZero
    @NotNull
    private Integer logBytesPerRequest;

    @PositiveOrZero
    @NotNull
    private Float authRatio;
}

File: frontend/src/features/project/types/request/update-project-assumption-request.ts
export type UpdateProjectAssumptionRequest = {
  concurrentUsers: number;
  headroom: number;
  requestsPerUserPerSecond: number;
  apiCallsPerRequest: number;
  dbRatioPerRequest: number;
  searchRatioPerRequest: number;
  cacheRatioPerRequest: number;
  kafkaRatioPerRequest: number;
  logBytesPerRequest: number;
  authRatio: number;
};
